### PR TITLE
display id updates, abacus -> 1.8.90

### PR DESCRIPTION
**Pull request by mike-dydx**
## Description / Intuition
- updates abacus
- `id` -> `displayId` changes
- removed prediction market test code
- fixed markets filter truncation

<br/>

## Before/After Screenshots or Videos

| Before | After |
|--------|-------|
|  |  |
|  |  |
|  |  |

<br/>

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring or Technical Debt
- [ ] Documentation update
- [ ] Other (please describe: )

### CODE REVIEW

File: dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/dydxMarketInfoViewBuilder.swift [hunk 1]
```diff
@@ -174,7 +174,7 @@ private class dydxMarketInfoViewPresenter: HostedViewPresenter<dydxMarketInfoVie
             guard let marketId = self?.marketId,
                   let assetId = marketMap[marketId]?.assetId,
                   let asset = assetMap[assetId] else { return }
-            if asset.tags?.contains("Prediction Market") != true && !Self.hidePredictionMarketsNotice {
+            if asset.tags?.contains("Prediction Market") == true && !Self.hidePredictionMarketsNotice {
```
Comment (mike-dydx) [inline dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/dydxMarketInfoViewBuilder.swift:177]: this is a change to stop forcing the prediction markets notice screen. Was forcing this for the test build.

File: dydx/dydxPresenters/dydxPresenters/_v4/Markets/Shared/SharedMarketPresenter.swift [hunk 2]
```diff
@@ -64,10 +63,15 @@ class SharedMarketPresenter: HostedViewPresenter<SharedMarketViewModel>, SharedM
                                                                     sign: market.priceChange24HPercent?.doubleValue ?? 0 >= 0 ? .plus : .minus,
                                                                     coloringOption: .allText)
         }
-        if let key = asset?.resources?.primaryDescriptionKey {
+        // sometimes the descriptions are unavailable, need to check localized output to ensure availability
+        if let key = asset?.resources?.primaryDescriptionKey,
+           DataLocalizer.localize(path: "APP.\(key)") != "APP.\(key)"
+        {
```
Comment (mike-dydx) [inline dydx/dydxPresenters/dydxPresenters/_v4/Markets/Shared/SharedMarketPresenter.swift:69]: prevents this:
IMG_4255939EAA06-1

File: dydx/dydxViews/dydxViews/_v4/CancelOrders/dydxCancelPendingIsolatedOrdersView.swift [hunk 3]
```diff
@@ -21,31 +21,31 @@ public class dydxCancelPendingIsolatedOrdersViewModel: PlatformViewModel {
     @Published public var state: State = .readyToSubmit
 
     @Published public var marketLogoUrl: URL?
-    @Published public var marketName: String
-    @Published public var marketId: String
+    @Published public var assetName: String
+    @Published public var assetId: String
```
Comment (mike-dydx) [inline dydx/dydxViews/dydxViews/_v4/CancelOrders/dydxCancelPendingIsolatedOrdersView.swift:25]: just a variable rename to more closely match abacus model property names

File: dydx/dydxViews/dydxViews/_v4/Markets/Shared/SharedMarketView.swift [hunk 4]
```diff
@@ -10,9 +10,8 @@ import PlatformUI
 import Utilities
 
 public final class SharedMarketViewModel: PlatformViewModel, Equatable {
-    @Published public var id: String?
```
Comment (mike-dydx) [inline dydx/dydxViews/dydxViews/_v4/Markets/Shared/SharedMarketView.swift:13]: this was unused

File: dydx/dydxViews/dydxViews/_v4/Markets/Shared/SharedMarketView.swift [hunk 5]
```diff
@@ -10,9 +10,8 @@ import PlatformUI
...
-    @Published public var tokenSymbol: String?
-    @Published public var tokenFullName: String?
```
Comment (mike-dydx) [inline dydx/dydxViews/dydxViews/_v4/Markets/Shared/SharedMarketView.swift:15]: another variable rename to more closely match abacus model property names

File: dydx/dydxViews/dydxViews/_v4/MarketInfo/Components/Resources/dydxMarketResourcesView.swift [hunk 6]
```diff
@@ -61,11 +61,11 @@ public class dydxMarketResourcesViewModel: PlatformViewModel {
                         HStack {
                             VStack(alignment: .leading) {
                                 HStack(spacing: 4) {
-                                    Text(sharedMarketViewModel.tokenFullName ?? "")
+                                    Text(sharedMarketViewModel.assetName ?? "")
                                         .themeColor(foreground: .textPrimary)
                                         .themeFont(fontType: .plus, fontSize: .medium)
                                 }
-                                Text(sharedMarketViewModel.tokenSymbol ?? "")
+                                Text(sharedMarketViewModel.assetId ?? "")
```
Comment (prashanDYDX) [inline dydx/dydxViews/dydxViews/_v4/MarketInfo/Components/Resources/dydxMarketResourcesView.swift:68]: confirming this is display id?

File: dydx/dydxViews/dydxViews/_v4/Markets/Shared/SharedMarketView.swift [hunk 7]
```diff
@@ -10,9 +10,8 @@ import PlatformUI
...
+    @Published public var assetId: String?
```
Comment (prashanDYDX) [inline dydx/dydxViews/dydxViews/_v4/Markets/Shared/SharedMarketView.swift:13]: is this display or raw id?

Comment (mike-dydx) [reply dydx/dydxViews/dydxViews/_v4/Markets/Shared/SharedMarketView.swift:13]: this gets set on line 52 of [this file](https://github.com/dydxprotocol/v4-native-ios/blob/cbc0c1169e0375a7c49a2291d7e567958cc14beb/dydx/dydxPresenters/dydxPresenters/_v4/Markets/Shared/SharedMarketPresenter.swift)

It's using `Asset.id` which is the token symbol (e.g. ETH) and assetName is the token (e.g Ethereum) whereas `PerpetualMarket.id` is the market (e.g. ETH-USD, or newly `ETH,raydium,F3nefJBcejYbtdREjui1T9DPh5dBgpkKq7u2GAAMXs5B-USD`) and `PerpetualMarket.displayId` would be `ETH-USD`

File: dydx/dydxViews/dydxViews/_v4/MarketInfo/Components/Resources/dydxMarketResourcesView.swift (hunk 6 above)
Comment (mike-dydx) [reply dydx/dydxViews/dydxViews/_v4/MarketInfo/Components/Resources/dydxMarketResourcesView.swift:68]: see [comment](https://github.com/dydxprotocol/v4-native-ios/pull/229/files#r1722316474)

File: dydx/dydxViews/dydxViews/_v4/Markets/Shared/SharedMarketView.swift (hunk 7 above)
Comment (mike-dydx) [reply dydx/dydxViews/dydxViews/_v4/Markets/Shared/SharedMarketView.swift:13]: additionally `PerpetualMarket.assetId` is the same as `Asset.id` (e.g.ETH)

**Review by prashanDYDX — APPROVED** [review]

**Pull request merged**